disc/main.py

```diff
@@ -0,0 +1,96 @@
+from flask import Flask, Response, abort, jsonify, request, session
+from flask.sessions import SecureCookieSessionInterface
+from flask_cors import CORS, cross_origin
+
+from common.db import connect_db, transaction_db
+from common.oauth_client import (
+    create_oauth_client,
+    get_user,
+    is_enrolled,
+    login,
+)
+
+VALID_ORIGINS = r"https://.*cs61a\.org"
+
+app = Flask(__name__, static_folder="", static_url_path="")
+if __name__ == "__main__":
+    app.debug = True
+
+CORS(app, origins=VALID_ORIGINS, supports_credentials=True)
+
+create_oauth_client(app, "61a-discussions")
+
+"""
+This is normally risky! It is safe here because 
+    (a) CORS prevents an attacker from reading reply data, even if their request is authenticated
+    (b) The endpoints require a Content-Type of application/json, so browsers will send a pre-flight
+        meaning that they will not even send POSTs unless they pass the CORS check, which they don't
+
+To ensure safety, we must ensure that 
+    (a) GET requests have no side-effects
+    (b) POST requests must have the JSON Content-Type, so a malicious site cannot send requests on
+        behalf of a user that can cause issues.
+    (c) The CORS policy only allows *.cs61a.org to send POSTs with credentials
+"""
+
+
+@app.after_request
+def patch_session_samesite(response: Response):
+    response.headers.add(
+        "Set-Cookie",
+        f"session={SecureCookieSessionInterface().get_signing_serializer(app).dumps(dict(session))}; "
+        "HttpOnly; SameSite=Lax; Path=/; Secure;",
+    )
+    return response
+
+
+with connect_db() as db:
+    db(
+        """CREATE TABLE IF NOT EXISTS saves (
+    email varchar(128),
```

Comment:
What happens if they change their email? (Is there an ID that is unchanging even if email changes?)

Reply:
We typically use emails as the unique key, since not everyone has an SID. For the handful of students who change their email, I guess I can add a simple mechanism to update their email in the db, like what we do for the other apps.